app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PicturesController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @pictures = Picture.all
+  end
+
+  def new
+    @picture = Picture.new
+  end
+
+  def picture_params
+    params.require(:picture).permit(:name, :image)
+  end
```

Comment:
I'd expect to see this at the bottom of the controller, maybe even in a `protected` or `private` section